format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -115,21 +113,27 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     PluginProperties.Builder builder = PluginProperties.builder();
     builder.addAll(config.getRawProperties().getProperties());
 
-    if (shouldGetSchema()) {
-      // Include source file system properties for schema auto detection
-      builder.add(FILE_SYSTEM_PROPERTIES, GSON.toJson(getFileSystemProperties(null)));
-    }
-
     ValidatingInputFormat validatingInputFormat =
       pipelineConfigurer.usePlugin(ValidatingInputFormat.PLUGIN_TYPE, fileFormat, fileFormat, builder.build());
     FormatContext context = new FormatContext(collector, null);
     validateInputFormatProvider(context, fileFormat, validatingInputFormat);
+    validatePathField(collector, schema);
 
-    if (validatingInputFormat != null && shouldGetSchema()) {
+    if (validatingInputFormat != null && schema == null && !config.containsMacro("schema")) {
```

Comment:
here we are skipping the schema detection if schema is not null. I remember UI will always have default schema with text format, so we are relying on the fact that UI does not pass in the schema for getSchema to work. Some of our db sources always get schema no matter the schema is passed in or not, is there a reason we don't always get schema here? Is it because we want to let user fix the schema if our schema detection is somehow wrong(since we only do a small sample to get the schema)?

Reply:
Is the schema not always null when the 'get schema' button is used?

If not, how is the plugin tell the difference between this is a schema that should be overridden by detection vs something the user wants to use? For example, with the delimited formats, the user will normally rename the columns to something more sensible.

Reply:
For DBs, we had similar issues where for compatibility reasons, the plugin needs to be able to read some columns as a different type. For example, before the datetime type was added, it would read those columns as strings. So we needed to add some logic that allowed somebody to specify a schema where datetime columns are strings.

Reply:
On a side note, I don't think the existing behavior works with the path field property either. Users are supposed to be able to add fields for the file path.

Reply:
👍  I think when get schema is hit, the schema will be null from UI. I just want to understand the use case here.